src/Steam.package/STOSSUserInfos.class/class/with.infos..st

```diff
@@ -0,0 +1,4 @@
+instance creation
+with: anObj infos: infos
+	anObj class = STZnGitHubOAuth2Session ifTrue: [ ^ self github: infos ].
+	anObj class = STZnGoogleOAuth2Session ifTrue: [ ^ self google: infos ].
```

Comment:
Ugly! You should just use polymorphism!

```
anObj newUserInfo: infos
```

Like that, if we add new AuthSessions, it should just work without adding a new if!

Also, please, choose better names for the variables and the methods!

What is the `infos` argument? Because the `github:` and `google:` method return also **Infos**, but they are not the same!

Why the methods are called `github:` and `google:` instead of `newGithubUserInformation:` and `newGoogleUserInformation` ?

If a method creates something => prefix it with `new`, `create`, `build`
If a method returns a boolean, prefix it with `is` or `has`. E.g, `isNice`, `hasChildren`
If a method will DO something (like writing to a database) the name should look like an order `writeToDatabase`, `saveIntoDatabase`, `eatThis`